crates/ruff/src/rules/wemake_python_styleguide/rules/too_short_name.rs

```diff
@@ -0,0 +1,156 @@
+
+/// ## What it does
+/// Checks for short variable or module names.
+///
+/// ## Why is this bad?
+/// It is hard to understand what the variable means and why it is used, if its name is too short.
+///
+/// ## Options
+/// - `wemake-python-styleguide.min-name-length`
+///
+/// ## Example
+/// ```python
+/// x = 1
+/// y = 2
+/// ```
+///
+/// Use instead:
+/// ```python
+/// x_coordinate = 1
+/// abscissa = 2
+/// ```
+#[violation]
+pub struct TooShortName {
+    pub name: String,
+    pub is_module: bool,
+}
+
+impl Violation for TooShortName {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TooShortName { name, is_module } = self;
+        format!(
+            "Found too short {}: `{}`",
+            if *is_module { "module name" } else { "name" },
+            name,
+        )
+    }
+}
+
+pub(crate) struct Checkable(ast::Identifier);
+
+impl From<&ast::Identifier> for Checkable {
+    fn from(value: &ast::Identifier) -> Self {
+        Self(value.clone())
+    }
+}
+
+impl From<&ast::Arg> for Checkable {
+    fn from(value: &ast::Arg) -> Self {
+        Self(value.arg.clone())
+    }
+}
+
+impl TryFrom<&ast::Alias> for Checkable {
+    type Error = ();
+
+    fn try_from(value: &ast::Alias) -> Result<Self, Self::Error> {
+        match value {
+            ast::Alias {
+                asname: Some(identifier),
+                ..
+            } => Ok(identifier.into()),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&ast::Expr> for Checkable {
+    type Error = ();
+
+    fn try_from(value: &ast::Expr) -> Result<Self, Self::Error> {
+        match value {
+            ast::Expr::Name(ast::ExprName { id, range, .. }) => {
+                Ok(Self(ast::Identifier::new(id, *range)))
+            }
+            ast::Expr::Attribute(ast::ExprAttribute { attr, .. }) => Ok(attr.into()),
+            _ => Err(()),
+        }
+    }
+}
+
+impl TryFrom<&Box<ast::Expr>> for Checkable {
+    type Error = ();
+
+    fn try_from(value: &Box<ast::Expr>) -> Result<Self, Self::Error> {
+        (&(**value)).try_into()
+    }
+}
+
+/// WPS111
+pub(crate) fn too_short_name(checker: &mut Checker, node: impl TryInto<Checkable>) {
+    if let Ok(Checkable(identifier)) = node.try_into() {
+        if naming::is_too_short_name(
+            identifier.as_str(),
+            checker.settings.wemake_python_styleguide.min_name_length,
+            true,
+        ) {
+            checker.diagnostics.push(Diagnostic::new(
+                TooShortName {
+                    name: identifier.to_string(),
+                    is_module: false,
+                },
+                identifier.range(),
+            ));
+        }
+    };
+}
+
+/// WPS111 (for filesystem)
+pub(crate) fn too_short_module_name(
+    path: &Path,
+    package: Option<&Path>,
+    settings: &Settings,
+) -> Option<Diagnostic> {
+    if !path
+        .extension()
+        .map_or(false, |ext| ext == "py" || ext == "pyi")
```

Comment:
```suggestion
    if !matches!(path.extension(), Some("py" | "pyi"))
```

Reply:
Pattern matching does not seem to work here, because `file_name` is an [OsStr](https://doc.rust-lang.org/std/ffi/struct.OsStr.html). Compiler says:

```
rustc: mismatched types
   expected reference `&std::ffi::OsStr`
      found reference `&'static str` [E0308]
```